cmd/gameserverapi/Dockerfile

```diff
@@ -1,19 +1,9 @@
 # build stage
-FROM golang:1.16 as builder
-
-WORKDIR /workspace
-# Copy the Go Modules manifests
-COPY go.mod ./
-COPY go.sum ./
-# cache deps before building and copying source so that we don't need to re-download as much
-# and so that source changes don't invalidate our downloaded layer
-RUN go mod download
-
-# Copy the go source
-COPY *.go ./
+ARG revision
+FROM thundernetes-src:$revision as builder
 
 # Build
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o thundernetes-gameserver-api-service 
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o thundernetes-gameserver-api-service ./cmd/gameserverapi
```

Comment:
(not relevant to your PR but just asking) what happens when we build this on M1 hardware? I'd assume this is emulated by default?

Reply:
go will build native for amd64. On the M1 Mac, if you run the application it will use Rosetta to do the emulation/conversion. This is happening already for docker all up, unless there is an Apple Silicon native version of docker now